src/Core/src/Platform/iOS/ApplicationExtensions.cs

```diff
@@ -35,14 +35,14 @@ public static void CreateNativeWindow(this UIApplicationDelegate nativeApplicati
 				dicts.Add(launchOptions);
 
 			var window = CreateNativeWindow(application, null, dicts.ToArray());
-			if (window is not null)
+			if (window is not null && nativeApplication is UIApplicationDelegate nativeAppDelegate)
 			{
-				nativeApplication.Window = window;
-				nativeApplication.Window.MakeKeyAndVisible();
+				nativeAppDelegate.Window = window;
+				nativeAppDelegate.Window?.MakeKeyAndVisible();
```

Comment:
I am wondering if we need this cast. Right now, this code might not actually work because it is NOT this type. I think with objc protocols, we need to use the interface. The reason they do not exist on the C# interfaces is because they are optional. So we should use the _Extensions type:

https://docs.microsoft.com/en-us/dotnet/api/uikit.uiapplicationdelegate_extensions.getwindow?view=xamarin-ios-sdk-12

Reply:
Great catch I missed it, yeah no need for the cast, you have to use the extensions ones, great catch @mattleibow I should not review during calls...

Reply:
Should be updated now